mcodingbot/plugins/highlights.py

```diff
@@ -1,22 +1,54 @@
 from __future__ import annotations
 
+import logging
 import asyncio
 from collections import defaultdict
 from typing import NamedTuple
 
 import crescent
 import hikari
+import toolbox
 from asyncpg import UniqueViolationError
 from floodgate import FixedMapping
 
 from mcodingbot.config import CONFIG
 from mcodingbot.database.models import Highlight, User, UserHighlight
 from mcodingbot.utils import Context, Plugin
 
+LOGGER = logging.getLogger(__file__)
+
 MAX_HIGHLIGHTS = 25
 MAX_HIGHLIGHT_LENGTH = 32
 
 
+async def has_permission(guild_id: int, user_id: int, channel_id: int) -> bool:
+    member = plugin.app.cache.get_member(guild_id, user_id)
+    if not member:
+        return False
+
+    channel = plugin.app.cache.get_guild_channel(channel_id)
+    if not channel:
```

Comment:
I'd consider using `:=` here

Reply:
i hate that here so i think thats why circuit didn't do that

Reply:
eh I think it's fine either way